src/main/java/com/google/gcp_variant_transforms/library/VariantToBqUtilsImpl.java

```diff
@@ -190,36 +194,40 @@ public void addFormatAndPhaseSet(TableRow row, Genotype genotype, VCFHeader vcfH
-      String fieldName = formatHeaderLine.getID();
-      if (fieldName.equals(VCFConstants.GENOTYPE_KEY)) {
+      String attrName = formatHeaderLine.getID();
+      if (attrName.equals(VCFConstants.GENOTYPE_KEY)) {
         continue; // We will set GT field in a separate "genotype" field in BQ row.
       }
-      if (genotype.hasAnyAttribute(fieldName)) {
-        Object value = genotype.getAnyAttribute(fieldName);
-        if (fieldName.equals(VCFConstants.GENOTYPE_ALLELE_DEPTHS) ||
-            fieldName.equals(VCFConstants.DEPTH_KEY) ||
-            fieldName.equals(VCFConstants.GENOTYPE_QUALITY_KEY) ||
-            fieldName.equals(VCFConstants.GENOTYPE_PL_KEY)) {
+      VCFFormatHeaderLine formatMetadata = vcfHeader.getFormatHeaderLine(attrName);
+      VCFHeaderLineType formatType = formatMetadata.getType();
+      if (genotype.hasAnyAttribute(attrName)) {
+        Object value = genotype.getAnyAttribute(attrName);
+        if (attrName.equals(VCFConstants.GENOTYPE_ALLELE_DEPTHS) ||
+            attrName.equals(VCFConstants.DEPTH_KEY) ||
+            attrName.equals(VCFConstants.GENOTYPE_QUALITY_KEY) ||
+            attrName.equals(VCFConstants.GENOTYPE_PL_KEY)) {
           // These four field values have been pre-processed in Genotype.
-          row.set(fieldName, value);
-        } else if (fieldName.equals(VCFConstants.PHASE_SET_KEY)) {
+          row.set(attrName, value);
+        } else if (attrName.equals(VCFConstants.PHASE_SET_KEY)) {
           String phaseSetValue = genotype.getAnyAttribute(VCFConstants.PHASE_SET_KEY).toString();
           phaseSet = replaceMissingWithNull(phaseSetValue);
         } else {
           // The rest of fields need to be converted to the right type as VCFHeader specifies.
-          VCFFormatHeaderLine formatMetadata = vcfHeader.getFormatHeaderLine(fieldName);
-          VCFHeaderLineType formatType = formatMetadata.getType();
           VCFHeaderLineCount formatCountType = formatMetadata.getCountType();
           if (formatCountType == VCFHeaderLineCount.INTEGER) {
-            row.set(fieldName, convertToDefinedType(genotype.getAnyAttribute(fieldName),
+            row.set(attrName, convertToDefinedType(genotype.getAnyAttribute(attrName),
                 formatType, formatMetadata.getCount()));
           } else {
             // If field number in the VCFHeader is ".", should pass a default count and do not
             // check if count is equal to the size of value.
-            row.set(fieldName, convertToDefinedType(genotype.getAnyAttribute(fieldName),
+            row.set(attrName, convertToDefinedType(genotype.getAnyAttribute(attrName),
                 formatType, Constants.DEFAULT_FIELD_COUNT));
           }
         }
+      } else if (formatType.equals(VCFHeaderLineType.Flag)) {
+        // If field is not presented in the VCF record and its field type is `Flag`, we need to
+        // set `false` in this field value.
+        row.set(attrName, false);
```

Comment:
We don't need to add Flag logic to `addFormatAndPhaseSet()`. FORMAT fields do not have Flag type:
https://samtools.github.io/hts-specs/VCFv4.3.pdf

Unless I am missing something here?

Please also update the unit tests to reflect this difference.

Reply:
Yeah, I also did not find any flag field in FORMAT so I just leave a "useless" check to handle potential bugs. I will remove this part and only check INFO field

Reply:
Please make sure you make this change before merging.